### PR TITLE
custom max_idletime in simple strategy

### LINKED ISSUE #1147: Configurable max_idletime on simple strategy
currently it is hard coded to 120s. It would be great to make this parameter configurable.

### PR DESCRIPTION
**Pull request by ZhuozhaoLi**
fixes #1147

### CODE REVIEW

File: parsl/config.py [hunk 1]
```diff
@@ -47,6 +47,8 @@ class Config(RepresentationMixin):
     strategy : str, optional
         Strategy to use for scaling resources according to workflow needs. Can be 'simple' or `None`. If `None`, dynamic
         scaling will be disabled. Default is 'simple'.
+    max_idletime : float, optional
+        The maximum idle time for an executor in the 'simple' strategy. Default is 120.0.
```
Comment (benclifford) [inline parsl/config.py]: specify units (seconds)

**Comment by benclifford** [comment]
whoops, I accidentally closed this.

File: parsl/config.py [hunk 2]
```diff
@@ -64,6 +66,7 @@ def __init__(self,
                  retries: int = 0,
                  run_dir: str = 'runinfo',
                  strategy: Optional[str] = 'simple',
+                 max_idletime: Optional[float] = 120.0,
```
Comment (benclifford) [inline parsl/config.py]: This type should be `float`, not `Optional[float]`.

`Optional[float]` means "this value may be `None` or `float`" - but `None` does not work with the simple strategy.

This is different from using the word "optional" to mean "there is a default float value so you do not need to explicitly specify one" - for example, look how the `usage_tracking` parameter is not `Optional[bool]`.

File: parsl/config.py (hunk 1 above)
Comment (ZhuozhaoLi) [reply parsl/config.py]: added

File: parsl/config.py (hunk 2 above)
Comment (ZhuozhaoLi) [reply parsl/config.py]: Good point. Fixed this.

**Pull request merged**